include/tvm/node/structural_equal.h

```diff
@@ -89,6 +91,15 @@ class StructuralEqual : public BaseValueEqual {
    * \return The comparison result.
    */
   TVM_DLL bool operator()(const ObjectRef& lhs, const ObjectRef& rhs) const;
+
+  /*!
+   * \brief Compare objects via strutural equal and save the path to the first mismatch.
+   *
+   * Same as operator(), but if `false` is returned, then `out_mismatch.lhs_path` and
+   * `out_mismatch.rhs_path` are set to the paths of the first mismatch.
+   */
+  bool FindFirstMismatch(const ObjectRef& lhs, const ObjectRef& rhs,
```

Comment:
I haven't implemented this one yet, so the only entry point right now is from python